src/Native/Unix/System.Security.Cryptography.Native.Apple/pal_x509.c

```diff
@@ -52,7 +63,21 @@ AppleCryptoNative_X509GetPublicKey(SecCertificateRef cert, SecKeyRef* pPublicKey
     if (cert == NULL || pPublicKeyOut == NULL || pOSStatusOut == NULL)
         return kErrorBadInput;
 
-    *pOSStatusOut = SecCertificateCopyPublicKey(cert, pPublicKeyOut);
+    pthread_once (&once, InitCertificateCopy);
+    // SecCertificateCopyPublicKey was deprecated in 10.14, so use SecCertificateCopyKey on the systems that have it (10.14+),
+    // and SecCertificateCopyPublicKey on the systems that don’t.
+    if (secCertificateCopyKey != NULL)
+    {
+        *pPublicKeyOut = (*secCertificateCopyKey)(cert);
+    }
+    else if (secCertificateCopyPublicKey != NULL)
+    {
+        *pOSStatusOut = (*secCertificateCopyPublicKey)(cert, pPublicKeyOut);
+    }
+    else
+    {
+        return kErrorBadInput;
```

Comment:
This is not really capturing the error (there is nothing wrong with the input), but I see we are ignoring the error code on the managed side in release builds anyway, so it does not really matter which error code we are going to return here. It may be worth a comment.

Reply:
It's not completely ignored, but anything that isn't 1 or 0 is all a generic CryptographicException.

Reply:
Here are the other options:

https://github.com/dotnet/corefx/blob/9908dee27b7a73046c8dae4a1db1b3603e795c8a/src/Native/Unix/System.Security.Cryptography.Native.Apple/pal_seckey.h#L12-L18

Reply:
"Unknown state" certainly sounds like what we'd be in if neither method exists.